src/components/SSOLanding/useSSOSession.js

```diff
@@ -0,0 +1,73 @@
+import { useEffect, useState } from 'react';
+import { useDispatch, useStore } from 'react-redux';
+import { useLocation } from 'react-router-dom';
+import { useCookies } from 'react-cookie';
+import queryString from 'query-string';
+
+import { config, okapi } from 'stripes-config';
+
+import { defaultErrors } from '../../constants';
+import { setAuthError } from '../../okapiActions';
+import { requestUserWithPerms } from '../../loginServices';
+import { parseJWT } from '../../helpers';
+
+const getParams = (location) => {
+  const search = location.search;
+
+  if (!search) return undefined;
+
+  return queryString.parse(search) || {};
+};
+
+const getToken = (cookies, params) => {
+  return cookies?.ssoToken || params?.ssoToken;
+};
+
+const getTenant = (params, token) => {
+  const tenant = config.useSecureTokens
+    ? params?.tenantId
+    : parseJWT(token)?.tenant;
+
+  return tenant || okapi.tenant;
+};
+
+const useSSOSession = () => {
+  const [isFailed, setIsFailed] = useState(false);
+  const store = useStore();
+  const dispatch = useDispatch();
+
+  const location = useLocation();
+  const [cookies] = useCookies(['ssoToken']);
+
+  const params = getParams(location);
+
+  const token = getToken(cookies, params);
+  const tenant = getTenant(params, token);
+
+  useEffect(() => {
+    requestUserWithPerms(okapi.url, store, tenant, token)
```

Comment:
@zburke @NikitaSedyx -- Or is it that this function is happy with `undefined` for `token` and will use the new httpOnly cookies in the request?

Reply:
Correct, @steveellis